planning/autoware_obstacle_cruise_planner/src/node.cpp

```diff
@@ -99,20 +118,48 @@ double calcDiffAngleAgainstTrajectory(
   return diff_yaw;
 }
 
+/**
+ * @brief Projects the obstacle's velocity onto the trajectory.
+ *
+ * This function calculates the obstacle's velocity components relative to the trajectory.
+ * It returns the longitudinal and perpendicular components of the obstacle's velocity
+ * with respect to the trajectory. Negative perpendicular velocity indicates that the
+ * obstacle is getting far away from the trajectory.
+ *
+ * @param traj_points The trajectory points.
+ * @param obstacle_pose The current pose of the obstacle.
+ * @param obstacle_twist The twist (velocity) of the obstacle.
+ * @return A pair containing the longitudinal and perpendicular velocity components.
+ */
 std::pair<double, double> projectObstacleVelocityToTrajectory(
```

Comment:
I guess we should change this function name since the significance of the function has changed.